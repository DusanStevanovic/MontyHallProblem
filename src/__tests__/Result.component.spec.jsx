// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Result from '../components/Result.component';
import type { SuccessObjectDataType } from '../types/index.type';

const mockProps: SuccessObjectDataType = {
  messages: ['Success !'],
  data: {
    win: 70,
    lose: 30,
    numberOfGames: 100,
    withChanging: true,
  }
}

describe('Result', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Result {...mockProps} />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should match progress with data from props for win game', () => {
    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper.find('.progressWin').prop('style')).toHaveProperty('width', '70%');
  });

  it('should match progress with data from props for lose game', () => {
    const wrapper = shallow(<Result {...mockProps} />);

    expect(wrapper.find('.progressLose').prop('style')).toHaveProperty('width', '30%');
  });
});