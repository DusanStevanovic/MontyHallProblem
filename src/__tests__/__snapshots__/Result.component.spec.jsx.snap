// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should match snapshot 1`] = `ShallowWrapper {}`;
